src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -1083,6 +1092,27 @@ DDLGuard::DDLGuard(Map & map_, std::shared_mutex & db_mutex_, std::unique_lock<s
     }
 }
 
+Names DatabaseCatalog::getAllRegisteredNames() const
+{
+    Names result;
+    auto database_instances = getDatabases();
+    for (const auto & database_name : database_instances | boost::adaptors::map_keys)
+    {
+        if (database_name == TEMPORARY_DATABASE)
+            continue;
+        DatabasePtr database = DatabaseCatalog::instance().tryGetDatabase(database_name);
+        if (!database)
+            continue;
+
+        for (auto table_it = database->getTablesIterator(getContext()); table_it->isValid(); table_it->next())
+        {
+            const auto & storage_id = table_it->table()->getStorageID();
+            result.emplace_back(storage_id.getFullTableName());
+        }
+    }
```

Comment:
I think it would be better to ad separate hints for database name and for table name

Reply:
That's good idea. I have fixed it.